gateways/istio-ingress/values.yaml

```diff
@@ -38,9 +38,9 @@ gateways:
       name: tls
 
     # Scalability tunning
-    # replicaCount: 1
+    # replicaCount: 2
     autoscaleEnabled: true
-    autoscaleMin: 1
```

Comment:
Doesn't seem like this needs to be commented out, or changed to 2. This affects autoscaling so it can be set to 1 and scale up to 2 fine

Reply:
I wouldn't expect the HPA and PDB logic play too well together... The former scales on load, the latter prevents scale to zero. If there is no load I'd expect HPA scales to autoscaleMin, if there is only one replica PDB prevents node drain (to keep that one instance).

Reply:
Another question... Is the default supposed to be highly available? @costinm?

If so we'd need an autoscaleMin of 3. Otherwise we would only account for planned downtime like node maintenance, not for a failure during planning downtime.

Reply:
Kubernetes won't scale up for the node drain? I am a bit suspicious of that, with a caveat that I have never tried it or seen docs on it, so it is mostly just a guess. 

Setting our default replicas to 2 is not a trivial change for an issue that may or may not exist for other users. At the least, can you provide some references that this is the right fix for this issue?

Reply:
I didn't try either. My expectation towards Kubernetes is just that they haven't coupled HPA and PDB.

Reply:
I can confirm it hadn't drained the HPA scaled pods after 2 hours on GKE 1.12.7 and GKE 1.13.6. I may have misconfigured something, but it was a pretty bog standard clean cluster created for the test, ditto istio install.

I may have had a pod without HPA on the node aswell and that prevented some scheduling logic, I will retest with only HPA.

Reply:
@howardjohn: After scanning through the PRs and issues related to introducing the PDB I'm less sure of what's the correct solution here. They were introduced to alleviated downtime during upgrade and that's a great use case. But i think it was a mistake to enable them by default.

Disruption budgets is to me conceptually something you layer on top of HA when disruption in planned maintenance windows also need to be mitigated. As it's set up today it might give false impression to users. I can't speak for anyone else but i think it's common to test involuntary disruption by simulating it through voluntary disruptions.

IE in the original issue that lead to default disruption budgets: https://github.com/istio/istio/issues/9951
It seems this user has only one replica running and it was critical for them not to interrupt istio during maintenance. I'd wager it's more critical for them not to have it interrupted outside maintenance, but that possibility may have been obscured for them.

The other option, HA by default, is not something i'd argue for. In addition to being, as @loewenstein said, 3+ replicas you'd also expect anti-affinity in there and that's rather hard to have good configs for given all possible cluster topologies.

Reply:
Yes, the default PDB will prevent k8s node drain if the replica if selected pod is 1, so we need to be careful of HPA and PDB combination. There was an issue for cert-manager, the default replica is 1 while minAvailable of PDB is also 1, that would prevent node upgrade/maintain... 

Maybe we need clear doc(at least inline comments) to end users about this.